shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureSurfaceProducer.java

```diff
@@ -0,0 +1,62 @@
+package io.flutter.embedding.engine.renderer;
+
+import android.graphics.SurfaceTexture;
+import android.media.Image;
+import android.view.Surface;
+import androidx.annotation.NonNull;
+import io.flutter.view.TextureRegistry;
+
+// An incomplete implementation of TextureRegistry.SurfaceProducer.
+//
+// TODO(https://github.com/flutter/flutter/issues/139702).
+final class SurfaceTextureSurfaceProducer
+    implements TextureRegistry.SurfaceProducer, TextureRegistry.ImageConsumer {
```

Comment:
This won't be an ImageConsumer, you'll have to introduce a new Consumer interface, maybe, "TextureRegistry.GLTextureConsumer".